src/grid.rs

```diff
@@ -193,6 +210,53 @@ impl<T> Grid<T> {
     }
 }
 
+impl<'a, T> Iterator for GridIterator<'a, T> {
+    type Item = &'a T;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let last_line = self.grid.num_lines() - Line(1);
+        let last_col = self.grid.num_cols() - Column(1);
+        match self.cur {
+            Point { line, col } if
```

Comment:
So I did some research on this, it doesn't look like we can escape the `if` blocks here because that is variable. It looks like under the hood, Rust `match`es patterns are static. You can do ranges, but it freaks out when you try to give it a variable to for the range bounds.